Makefile

```diff
@@ -561,6 +561,21 @@ nuke: clean clean-docker ## Clean plus removes persistant server data.
 setup-mac: ## Adds macOS hosts entries for Docker.
 	echo $$(boot2docker ip 2> /dev/null) dockerhost | sudo tee -a /etc/hosts
 
+update-dependencies: ## Uses go get -u to update all the dependencies while holding back any that require it. 
+	@echo Updating Dependencies
+
+	# Update all dependencies (does not update across major versions)
+	go get -u
+
+	# Keep back because of breaking API changes
+	go get -u github.com/segmentio/analytics-go@2.1.1
+
+	# Tidy up
+	go mod tidy
+
+	# Copy everything to vendor directory
+	go mod vendor
```

Comment:
The reason for vendoring is to ensure that we can always build any commit of the server without relying on external services to stay alive.

Reply:
I see. This is mostly a concern for CI/CD/, isn't it? Developers only have connect to the external services once after a dependency update.

I could see two different method to fix this problem:
- Rely on proxies for modules. This is the way the go team proposes. See https://github.com/golang/go/wiki/Modules#are-there-always-on-module-repositories-and-enterprise-proxies.
- Cache `$GOPATH/pkg/mod` over multiple pipeline runs

But to be fair: The only real problem with vendoring is that you have a copy of code that is already somewhere else.

I'm at most 2/5 on not vendoring. What do you think of the two proposed alternatives?

Reply:
As another data point, vendoring makes it extremely clear how to hack on the dependencies during testing. I just leveraged this to debug an issue in a vendored dependency by simply modifying the code in `vendor`. I imagine this is possible with go modules, though it would require substantially more setup.